src/Extensions/SearchFormFactoryExtension.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace SilverStripe\SearchService\Extensions;
+
+use SilverStripe\Assets\Image;
+use SilverStripe\Control\RequestHandler;
+use SilverStripe\Core\Extension;
+use SilverStripe\Forms\CheckboxField;
+use SilverStripe\Forms\DatetimeField;
+use SilverStripe\Forms\Form;
+use SilverStripe\Forms\FormFactory;
+
+class SearchFormFactoryExtension extends Extension
+{
+
+    public function updateForm(
+        Form $form,
+        ?RequestHandler $controller = null,
+        string $name = FormFactory::DEFAULT_NAME,
+        array $context = []
+    ): void {
+        $fields = $form->Fields()->findOrMakeTab('Editor.Details');
+        $file = $context['Record'] ?? null;
+
+        if (!$fields || !$file || $file instanceof Image) {
```

Comment:
could we make the class check configurable? I'm thinking that having images excluded by default is good, but if you wanted to for some reason you'd need a whole new extension. Maybe move it to a configurable array so people could modify it

Reply:
Fair point. Added configuration for `exclude_classes` and added extra for  `exclude_file_extensions` excluding file types